pkg/api/handlers/libpod/quadlets.go

```diff
@@ -238,3 +239,115 @@ func InstallQuadlets(w http.ResponseWriter, r *http.Request) {
 
 	utils.WriteResponse(w, http.StatusOK, installReport)
 }
+
+// RemoveQuadlet handles DELETE /libpod/quadlets/{name} to remove a quadlet file
+func RemoveQuadlet(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+
+	query := struct {
+		Force         bool `schema:"force"`
+		Ignore        bool `schema:"ignore"`
+		ReloadSystemd bool `schema:"reload-systemd"`
+	}{
+		ReloadSystemd: true, // Default to true like CLI
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusBadRequest, fmt.Errorf("failed to parse parameters for %s: %w", r.URL.String(), err))
+		return
+	}
+
+	name := utils.GetName(r)
+	if name == "" {
+		utils.Error(w, http.StatusBadRequest, fmt.Errorf("quadlet name must be provided"))
+		return
+	}
+
+	containerEngine := abi.ContainerEngine{Libpod: runtime}
+	removeOptions := entities.QuadletRemoveOptions{
+		Force:         query.Force,
+		Ignore:        query.Ignore,
+		ReloadSystemd: query.ReloadSystemd,
+	}
+
+	removeReport, err := containerEngine.QuadletRemove(r.Context(), []string{name}, removeOptions)
+	if err != nil {
+		// For systemd connection errors and other internal errors
+		utils.InternalServerError(w, err)
+		return
+	}
+
+	// Check if there are errors in the report for this specific quadlet
+	if err, ok := removeReport.Errors[name]; ok {
+		// If ignore=false and quadlet not found, return 404
+		if !query.Ignore && strings.Contains(err.Error(), "no such") {
+			utils.Error(w, http.StatusNotFound, fmt.Errorf("no such quadlet: %s: %w", name, err))
+			return
+		}
+		// If force=false and quadlet is running, return 400
+		if !query.Force && strings.Contains(err.Error(), "running") {
+			utils.Error(w, http.StatusBadRequest, err)
+			return
+		}
+	}
+
+	utils.WriteResponse(w, http.StatusOK, removeReport)
+}
+
+// RemoveQuadlets handles DELETE /libpod/quadlets to remove quadlet files (batch operation)
+func RemoveQuadlets(w http.ResponseWriter, r *http.Request) {
```

Comment:
It looks like `RemoveQuatlets` and `RemoveQuadlet` are redundant; we should consider consolidating them.

Reply:
reason why id didn't is to make a call to `DELETE /libpod/quadlets/{name}` + args and `DELETE /libpod/quadlets` + args, and I didn't like to handle the logic of name / no name in the function. I could do it, although this to me seemed more RESTful